tests/testthat/test-report.R

```diff
@@ -0,0 +1,88 @@
+context("report")
+
+report_test <- function() {
+    dat <- data.frame(
+        V1 = c(1, 2, 3),
+        V2 = c("a", "b", "c")
+    )
+
+    report <- data_validation_report()
+
+    validate(dat)  %>%
+        validate_cols(is.numeric)  %>%
+        validate_if(.data$V1 > 0)  %>%
+        add_results(report)
+
+    return(report)
+}
+
+get_advanced_view <- function(log) {
+    lines <- log[grep("type", log):length(log)]
+    lines <- lines[!grepl("^[[:blank:]+-=:_|]*$", lines)]
+    out <- t(
+        as.data.frame(
+            strsplit(gsub(" ", "", lines), "\\|"),
+            row.names = 1
+        )
+    )
+    out <- data.frame(out)
+
+    return(out)
+}
+
+test_that("exported CSV matched results obtained from get_results", {
+    tmp <- file.path(tempdir(), "test.csv")
+    on.exit(unlink(tmp))
+
+    report <- report_test()
+    save_results(report, tmp, method = write)
+
+    actual <- read.csv(tmp) %>%
+        dplyr::select(-"X") %>%
```

Comment:
That seems like an extra step required. Thus exported CSV does not match the `get_results`. Maybe it can be removed via some `read.csv` option?

Reply:
hello @mdubel,
I attempted to use different `method` on `save_results`, however, it doesn't seem to work due to a bug that I have already reported [here](https://github.com/Appsilon/data.validator/issues/64). I can't find any arguments from `read.csv` which allows to remove specific column, thus i added the extra step to remove the generated row ID. do you have any suggestion?

Reply:
I see. `save_results` uses the `write.csv` with the `row.names`, thus this `X` is preserved.